FTBS with gcc 14.2: error: format '%p' expects argument of type 'void *', but argument 3 has type 'JSVarRef *' [-Werror=format=]

Hi!

I tried to build quickjs-ng as a submodule in the my cmake project with GCC 14.2 and got errors (it looks like due to -Werror in the CMakeLists.txt):

```
.../quickjs.c: In function 'js_create_module_bytecode_function':
.../quickjs.c:27136:35: error: format '%p' expects argument of type 'void *', but argument 3 has type 'JSVarRef *' [-Werror=format=]
27136 |                 module_trace(ctx, "local %d: %p\n", i, var_ref);
      |                                   ^~~~~~~~~~~~~~~~     ~~~~~~~
      |                                                        |
      |                                                        JSVarRef *
.../quickjs.c:27044:15: note: in definition of macro 'module_trace'
27044 |        printf(__VA_ARGS__); \
      |               ^~~~~~~~~~~
.../quickjs.c:27136:47: note: format string is defined here
27136 |                 module_trace(ctx, "local %d: %p\n", i, var_ref);
      |                                              ~^
      |                                               |
      |                                               void *
.../quickjs.c: In function 'js_inner_module_linking':
.../quickjs.c:27347:39: error: format '%p' expects argument of type 'void *', but argument 2 has type 'JSVarRef *' [-Werror=format=]
27347 |                     module_trace(ctx, "local export (var_ref=%p)\n", var_ref);
      |                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~
      |                                                                      |
      |                                                                      JSVarRef *
.../quickjs.c:27044:15: note: in definition of macro 'module_trace'
27044 |        printf(__VA_ARGS__); \
      |               ^~~~~~~~~~~
.../quickjs.c:27347:63: note: format string is defined here
27347 |                     module_trace(ctx, "local export (var_ref=%p)\n", var_ref);
      |                                                              ~^
      |                                                               |
      |                                                               void *
cc1: all warnings being treated as errors
```

--- a/quickjs.c
+++ b/quickjs.c
@@ -27133,7 +27133,7 @@ static int js_create_module_bytecode_function(JSContext *ctx, JSModuleDef *m)
                 if (!var_ref)
                     goto fail;
 
-                module_trace(ctx, "local %d: %p\n", i, var_ref);
+                module_trace(ctx, "local %d: %p\n", i, (void *)var_ref);
 
                 var_refs[i] = var_ref;
             }
@@ -27344,7 +27344,7 @@ static int js_inner_module_linking(JSContext *ctx, JSModuleDef *m,
                     var_ref->header.ref_count++;
                     var_refs[mi->var_idx] = var_ref;
 
-                    module_trace(ctx, "local export (var_ref=%p)\n", var_ref);
+                    module_trace(ctx, "local export (var_ref=%p)\n", (void *)var_ref);
                 }
             }
         }

